pkg/pulumiyaml/ast/expr.go

```diff
@@ -141,7 +146,7 @@ type InterpolateExpr struct {
 func (n *InterpolateExpr) String() string {
 	var str strings.Builder
 	for _, p := range n.Parts {
-		str.WriteString(strings.ReplaceAll(p.Text, "$", "$$"))
+		str.WriteString(strings.ReplaceAll(p.Text, "$$", "$"))
```

Comment:
InterpolateExpr.String is going from the runtime representation back to the textual representation. We should be re-escaping here, not double unescaping.

Reply:
I reverted the change in this function to the original implementation and added a comment explaining what it was doing ✅